glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py

```diff
@@ -46,6 +46,9 @@ def kinetic_matrix(
             matrix_implementation,
         )
 
+        if megacomplex_scales is not None:
+            this_matrix *= megacomplex_scales[i]
```

Comment:
```suggestion
            this_matrix *= megacomplex_scales[k_matrix_index]
```
That LGTM won't complain about the use of `i` in nested for loops and it is more apparent which `i` does what.

Reply:
Added in commit "Avoid using same variable name in nested for loops"